src/utils/apiResponse.utils.ts

```diff
@@ -0,0 +1,29 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+export type ControllerResponse = {
+  isSuccess: number;
+  statusCode: number;
+  message: string;
+  data?: any;
+  error?: any;
+};
+
+export const APIResponse = {
+  create: (
+    isSuccess: number,
+    statusCode: number,
+    message?: string,
+    data?: any,
+    error?: any,
+  ): ControllerResponse => {
+    const response: ControllerResponse = { isSuccess, statusCode, message };
+    if (error) {
+      response.error = error;
+    } else if (!error && data) {
+      response.data = data;
+    } else {
+      response.data = {};
+    }
```

Comment:
PullProofAI

The `APIResponse` function can be refactored to improve readability and maintainability by simplifying the conditional logic.
```diff
export const APIResponse = {
  ): ControllerResponse => {
    const response: ControllerResponse = { isSuccess, statusCode, message, data: data || {}, error: error || undefined };

    if (error) {
      response.error = error;
    } else if (data) {
      response.data = data;
    }

    return response;
  },
```


<!-- This is an auto-generated comment by PullProofAI -->